File: Maid.Content.DB/Entity/Content/ContentSource.cs
using Maid.Core;

namespace Maid.Content.DB
{
	public class ContentSource : BaseLookup
	{
		public string DomainUrl { get; set; }

		public string ImageUrl { get; set; }

		public string TitleXpath { get; set; }

		public string ImageXpath { get; set; }

		public string StatusXpath { get; set; }

		public string CollectionItemXpath { get; set; }

		public string CollectionItemHrefXpath { get; set; }

		public string CollectionItemTitleXpath { get; set; }

		public string CollectionItemDateXpath { get; set; }
	}
}

File: UI/src/app/entity/content/content-source.ts
import { BaseLookup } from '../base-lookup';

export class ContentSource extends BaseLookup {

	domainUrl: string;
	imageUrl: string;
	
	titleXpath: string;
	imageXpath: string;

	collectionItemXpath?: string;
	collectionItemTitleXpath?: string;
	collectionItemDateXpath?: string;
	collectionItemHrefXpath?: string;

	statusXpath?: string;
	positiveStatusText?: string;

	constructor(entity?) {
		super(entity);
	}
}
